lib/pos_epoch_complete_hook.go

```diff
@@ -72,45 +92,95 @@ func (bav *UtxoView) RunEpochCompleteHook(blockHeight uint64) error {
 		return errors.New("RunEpochCompleteHook: CurrentEpochEntry is nil, this should never happen")
 	}
 
-	currentGlobalParamsEntry := bav.GetCurrentGlobalParamsEntry()
-
 	// Snapshot the current GlobalParamsEntry.
-	bav._setSnapshotGlobalParamsEntry(bav.GlobalParamsEntry, currentEpochEntry.EpochNumber)
+	bav._setSnapshotGlobalParamsEntry(bav.GetCurrentGlobalParamsEntry(), currentEpochEntry.EpochNumber)
```

Comment:
I might move all this snapshotting logic to a separate function. This way we can unit test the snapshotting logic if we want and will help keep this function nice and short. I can see us adding more snapshotting at some point and would be good to confine that to its own function.

Reply:
Sure. I'll move each of the three categories of operations into their own top-level functions